clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/userdata/LocalFileUserDataProvider.groovy

```diff
@@ -132,7 +151,7 @@ class LocalFileUserDataProvider implements UserDataProvider {
       .replace('%%cluster%%', cluster)
       .replace('%%stack%%', stack)
       .replace('%%detail%%', detail)
-      .replace('%%launchconfig%%', launchConfigName)
+      .replace(userDataRequest.launchTemplate ? '%%launchtemplate%%' : '%%launchconfig%%', userDataRequest.launchSettingName)
```

Comment:
sent you a message directly about the implications of this change on our templates